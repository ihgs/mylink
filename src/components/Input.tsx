import { ChangeEvent } from "react";

const Input = ({
  name,
  type = "text",
  value,
  placeholder,
  required = false,
  setValue,
}: {
  name: string;
  type?: string;
  value: string;
  placeholder: string;
  required?: boolean;
  setValue: any;
}) => {
  const onChange = (e: ChangeEvent<HTMLInputElement>) => {
    setValue(e.target.value);
  };

  return (
    <input
      type={type}
      name={name}
      value={value}
      onChange={onChange}
      className="input input-bordered w-full my-1"
      placeholder={placeholder}
      required={required}
    ></input>
  );
};

export default Input;
