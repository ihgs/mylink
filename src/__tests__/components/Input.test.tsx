import { render } from "@testing-library/react";
import Input from "../../components/Input";

test("[snapshot]Input", () => {
  const view = render(
    <Input
      name="test"
      value=""
      placeholder=""
      setValue={(value) => {
        console.log(value);
      }}
    ></Input>
  );
  expect(view.container).toMatchSnapshot();
});
