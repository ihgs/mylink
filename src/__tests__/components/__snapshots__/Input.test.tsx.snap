// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[snapshot]Input 1`] = `
<div>
  <input
    class="input input-bordered "
    name="test"
    placeholder=""
    type="text"
    value=""
  />
</div>
`;
